src/main/java/com/openlattice/shuttle/dates/JavaDateTimeHelper.java

```diff
@@ -1,81 +1,123 @@
 package com.openlattice.shuttle.dates;
 
 import com.openlattice.shuttle.util.Cached;
+import com.openlattice.shuttle.util.Constants;
 import org.apache.commons.lang3.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.sql.Timestamp;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.LocalTime;
 import java.time.OffsetDateTime;
 import java.time.format.DateTimeFormatter;
 import java.time.format.DateTimeParseException;
+import java.util.Optional;
 import java.util.TimeZone;
 import java.util.concurrent.ExecutionException;
 import java.util.function.BiFunction;
 
 public class JavaDateTimeHelper {
-    private static final Logger logger = LoggerFactory.getLogger(JavaDateTimeHelper.class);
+    private static final Logger logger = LoggerFactory.getLogger( JavaDateTimeHelper.class );
 
-    private final TimeZone      tz;
-    private final String[]      datePatterns;
+    private final Optional<TimeZone> tz;
+    private final String[]           datePatterns;
 
-    public JavaDateTimeHelper(TimeZone tz, String... datePatterns) {
+    public JavaDateTimeHelper( Optional<TimeZone> tz, String... datePatterns ) {
         this.tz = tz;
         this.datePatterns = datePatterns;
     }
 
-    private boolean shouldIgnoreValue(String date) {
-        return date == null || StringUtils.isBlank(date) || date.equals("NULL");
+    private boolean shouldIgnoreValue( String date ) {
+        return date == null || StringUtils.isBlank( date ) || date.equals( "NULL" );
     }
 
-    public <R> R parseWithTwoDigitYearHandling(String date, BiFunction<String, DateTimeFormatter, R> parseFunction,
+    public <R> R parseWithTwoDigitYearHandling(
+            String date, BiFunction<String, DateTimeFormatter, R> parseFunction,
             BiFunction<R, String, R> postParseFunction ) {
-        if (shouldIgnoreValue(date)) {
+        if ( shouldIgnoreValue( date ) ) {
             return null;
         }
-        for ( String datePattern: datePatterns ){
+        for ( String datePattern : datePatterns ) {
             try {
                 DateTimeFormatter formatter = Cached.getDateFormatForString( datePattern );
                 R result = parseFunction.apply( date, formatter );
                 return postParseFunction.apply( result, datePattern );
-            } catch ( DateTimeParseException e) {
-                if ( datePattern.equals(datePatterns[datePatterns.length - 1]) ) {
-                    logger.error("Unable to parse date {}, please see debug log for additional information: {}.", date,e);
+            } catch ( DateTimeParseException e ) {
+                if ( datePattern.equals( datePatterns[ datePatterns.length - 1 ] ) ) {
+                    logger.error( "Unable to parse date {}, please see debug log for additional information: {}.",
```

Comment:
Why does this log only happen if it's the last datePattern in the array?

Reply:
because if we specify multiple patterns, having at least one successful parser is enough.  If all patterns are tried, still not successful, it should throw the warning.

Reply:
gotcha, ty for clarifying!